src/main/java/io/gravitee/resource/oauth2/generic/OAuth2GenericResource.java

```diff
@@ -356,4 +365,44 @@ public void handle(Throwable event) {
     public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
         this.applicationContext = applicationContext;
     }
+
+    private ProxyOptions getSystemProxyOptions() {
```

Comment:
I start to see a lot of duplicates of this system proxy creation and I'm a bit concerned about the maintainability. Do you think this could be provided by a kind a util class located in gravitee-common ?

Reply:
Yes, actually I've just duplicated from a commit on AM resource to bootstrap the PR and thought about it but was wondering were it should sit :) I'll have a look at the other modules using this code and add something in gravitee-common then